JustSaying.IntegrationTests/Fluent/IntegrationTestBase.cs

```diff
@@ -90,26 +90,23 @@ protected async Task WhenAsync(IServiceCollection services, Func<IMessagePublish
             IMessagePublisher publisher = serviceProvider.GetRequiredService<IMessagePublisher>();
             IMessagingBus listener = serviceProvider.GetRequiredService<IMessagingBus>();
 
-            using (var source = new CancellationTokenSource(Timeout))
+            using (var cts = new CancellationTokenSource())
             {
-                try
-                {
-                    var delayTask = Task.Delay(Timeout, source.Token);
-                    var actionTask = action(publisher, listener, source.Token);
-
-                    await Task.WhenAny(actionTask, delayTask).ConfigureAwait(false);
+                var delayTask = Task.Delay(Timeout, cts.Token);
```

Comment:
The much more trivial change would be to change this to:
```csharp
var delayTask = Task.Delay(-1, cts.Token);
```

Reply:
However I find that very slightly more confusing because you won't be sure which of the 2 tasks from `WhenAny` was the one that finished first. Even with the code that follows it, it looks like a deliberate race condition.